src/Aspect/ContractCheckerAspect.php

```diff
@@ -156,7 +164,15 @@ private function isContractSatisfied($instance, $scope, array $args, $annotation
         }
         $instance = is_object($instance) ? $instance : null;
 
-        return $invoker->bindTo($instance, $scope)->__invoke($args, $annotation->value);
+        try {
+            $invocationResult = $invoker->bindTo($instance, $scope)->__invoke($args, $annotation->value);
+        } catch (\Exception $e) {
+            throw new ContractViolation($this->invocation, $annotation->value . " Details: " . $e->getMessage());
+        }
+
+        // if $invocationResult is null, $annotation->value didn't throw any exception
+        // for example - assertion passed (and didn't return bool value)
+        return $invocationResult === null || $invocationResult === true;
```

Comment:
Because there are four basic scenarios:
1. $invocationResult is boolean, so we must ensure it's true
2. $invocationResult raise Exception, so it's catched
3. $invocationResult **is null**, so there was executed for example assertion from beberlei/assert - it didn't throw Exception, so it passed
4. any other results (1, strings and so on) shouldn't be supported